src/awala.spec.ts

```diff
@@ -97,4 +101,45 @@ describe('awala', () => {
       expect(countResult).toBe(1);
     });
   });
+
+  describe('postToAwala', () => {
+    const mockFetch = mockSpy(jest.spyOn(global, 'fetch'));
+    const TEST_AWALA_ENDPOINT : URL = new URL(AWALA_MIDDLEWARE_ENDPOINT)
+    const TEST_RECIPIENT_ID : string = "TEST_RECIPIENT_ID"
+    const contentTypeHeaderName = 'content-type';
+    const awalaRecipientHeaderName = 'X-Awala-Recipient';
+    const awalaPostData = "Test data";
+
+    test("Should send data to Awala", async () => {
+      mockFetch.mockResolvedValue(new Response(JSON.stringify({ recipientId: TEST_RECIPIENT_ID})));
+
+      const awalaResponse = await postToAwala(awalaPostData,AWALA_PDA,TEST_AWALA_ENDPOINT);
+
+      expect(awalaResponse.didSucceed).toBeTrue();
+      expect(mockFetch).toHaveBeenNthCalledWith(1, TEST_AWALA_ENDPOINT, {
+        method: 'POST',
+        headers: { [contentTypeHeaderName]: 'application/vnd+relaycorp.awala.pda-path' },
+        body: AWALA_PDA,
+      })
+      expect(mockFetch).toHaveBeenNthCalledWith(2, TEST_AWALA_ENDPOINT, {
+        method: 'POST',
+        headers: {
+          [contentTypeHeaderName]: 'application/vnd.veraid.member-bundle',
+          [awalaRecipientHeaderName]: TEST_RECIPIENT_ID
+        },
+        body: awalaPostData,
+      })
+    })
```

Comment:
Since the test is pretty small I made just one test for it. Should we split this into 2 parts? 
1 - should make authorization request
2 - should post data

Reply:
Yes, please. I think it should be split. I should be able to understand the requirements from the test suite.